presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/TarController.java

```diff
@@ -0,0 +1,235 @@
+package com.teradata.prestomanager.agent;
+
+import com.google.inject.Inject;
+import io.airlift.log.Logger;
+
+import javax.ws.rs.client.Client;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.teradata.prestomanager.agent.AgentFileUtils.downloadFile;
+import static java.lang.String.format;
+import static java.nio.file.Files.createDirectories;
+import static java.nio.file.Files.delete;
+import static java.nio.file.Files.isDirectory;
+import static java.nio.file.Files.isRegularFile;
+import static java.util.Objects.requireNonNull;
+
+public class TarController
+        extends PackageController
+{
+    private static final Logger LOGGER = Logger.get(RpmController.class);
+
+    private final Path configDir;
+    private final Path catalogDir;
+    private final Path dataDir;
+    private final Path pluginDir;
+    private final CommandExecutor executor;
+    private final PrestoConfigDeployer configUtils;
```

Comment:
I mean that it's not a utility class in the traditional sense.  it's a standard java object that has state and does stuff.  And it should be called by its more descriptive name of configDeployer so people reading the code have a sense of what it does.